backend/backend.go

```diff
@@ -27,6 +29,22 @@ type Backend interface {
 	State() (state.State, error)
 }
 
+// MultiState is an interface that a backend can implement to allow changing
+// between named states depending on the configured environment.
+type MultiState interface {
```

Comment:
It may be, but I also think an interface should represent the full set of functionality required to make it useful. In the context of Terraform, I believe that is loading AND changing states, so splitting it into two required interface seems more of a mental burden than having one required interface. 

So I think it should be part of the one interface.